### PR TITLE
LULCC + CROP

### PR DESCRIPTION
**Pull request by tanzeli1982**
Fix bugs that break the coupling of the dynamic land use model and the crop model. 

[BFB]

### CODE REVIEW

**Comment by tanzeli1982** [comment]
@bishtgautam 
For the new test, I downloaded the LULC file from https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_1x1_smallvilleIA_mp24_hist_simyr1850-1855_c141229.nc. It is also available on Compy (/compyfs/inputdata/lnd/clm2/surfdata_map/landuse.timeseries_1x1_smallvilleIA_mp24_hist_simyr1850-1855_c141229.nc).

**Comment by bishtgautam** [comment]
@tanzeli1982, Can you please rebase this branch to resolve conflict? thanks.

**Comment by tanzeli1982** [comment]
> @tanzeli1982, Can you please rebase this branch to resolve conflict? thanks.

Done

**Comment by thorntonpe** [comment]
@tanzeli1982 Can you summarize the format changes that were required in VegetationDataType.F90? the diff is too large to load in github.

**Comment by tanzeli1982** [comment]
> @tanzeli1982 Can you summarize the format changes that were required in VegetationDataType.F90? the diff is too large to load in github.

In my old submission, there is a special character "^M" added to each line of VegetationDataType.F90. It was corrected by my new submission. Now, you should be able to see the true differences in github.

**Comment by thorntonpe** [comment]
@tanzeli1982 I see that the PR is labeled BFB. Just checking: is this passing the full suite of land developer tests BFB? and is it still BFB when the crop model is turned on?

**Comment by tanzeli1982** [comment]
> @tanzeli1982 I see that the PR is labeled BFB. Just checking: is this passing the full suite of land developer tests BFB? and is it still BFB when the crop model is turned on?

It has passed the full land developer tests BFB. Is it still BFB when the crop model is turned on? I do not fully understand what you mean. If it refers to SMS_Ly2_P1x1.1x1_smallvilleIA.ICLM45CNCROP.compy_intel.clm-force_netcdf_pio which is the only test with crop model on, yes, the results are BFB.

**Comment by thorntonpe** [comment]
Great, thanks. I need a little more time to look at the code changes, but will finalize my review by the end of the week.

**Comment by tanzeli1982** [comment]
Hello all, do we have progress on this PR?

**Comment by tanzeli1982** [comment]
@bishtgautam Should we merge it since no new comments come out?

**Comment by tanzeli1982** [comment]
@thorntonpe Is the code in a good shape?

**Comment by rljacob** [comment]
@thorntonpe any comments on this PR?

**Comment by tanzeli1982** [comment]
@rljacob @bishtgautam How should we handle this PR?

**Comment by rljacob** [comment]
We are waiting for @thorntonpe to finalize his review.
